doc/threat_model.md

```diff
@@ -0,0 +1,111 @@
+# td-shim threat model and secure design
+
+[TDX reference](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-trust-domain-extensions.html)
+
+## Trust Computing Base (TCB)
+
+B-1. TDX-Module and its SOC CPU are trusted.
+
+The result of TDCALL (except TDVMCALL) can be trusted.
+
+B-2. Architecture TD or service TD is trusted.
+
+The architecture TD and service TD SHALL support attestation.
+
+B-3. VMM, Legacy VM or other user TD are not trusted.
+
+The result of TDVMCALL cannot be trusted.
```

Comment:
An ASCII art diagram like the one in `design.md` would be great here to show the trust boundaries.

Reply:
will add